pkg/provider/secretserver/client.go

```diff
@@ -62,36 +62,30 @@ func (c *client) GetSecret(ctx context.Context, ref esv1.ExternalSecretDataRemot
 		return jsonStr, nil
```

Comment:
we will still end up going to this return here. I don't think your implementation is correct for what you are trying to achieve.